### PR TITLE
Create blockfrost provider

### PR DESCRIPTION
**Pull request by bernokl**
Implemented as part of https://input-output.atlassian.net/browse/LW-11745

### CODE REVIEW

File: nix/cardano-services/deployments/blockfrost-backend.provider.nix [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        ASSET_PROVIDER = "blockfrost";
+        CHAIN_HISTORY_PROVIDER = "blockfrost";
+        NETWORK_INFO_PROVIDER = "blockfrost";
+        REWARDS_PROVIDER = "blockfrost";
+        UTXO_PROVIDER = "blockfrost";
+        BLOCKFROST_API_KEY = {
+          valueFrom.secretKeyRef = {
+            name = "blockfrost";
+            key = "api-key";
+          };
```
Comment (iccicci) [inline nix/cardano-services/deployments/blockfrost-backend.provider.nix:45]: Do we have a single token which is valid for all the networks, or should this be improved a bit to pick the correct token based on `config.network`?

**Review by iccicci — DISMISSED** [review]
Apart for a minor question, LGTM!

File: nix/cardano-services/deployments/blockfrost-backend-ingress.nix [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  lib,
+  values,
+  config,
+  utils,
+  ...
+}: {
+  templates.accelerator = lib.mkIf (values.blockfrost-backend.useAccelerator && values.ingress.enabled) {
+    apiVersion = "operator.h3poteto.dev/v1alpha1";
+    kind = "EndpointGroupBinding";
+    metadata.name = "${config.name}-main";
```
Comment (gytis-ivaskevicius) [inline nix/cardano-services/deployments/blockfrost-backend-ingress.nix:11]: This will create a conflict

**Review by gytis-ivaskevicius — APPROVED** [review]

File: nix/cardano-services/deployments/blockfrost-backend-ingress.nix (hunk 2 above)
Comment (bernokl) [reply nix/cardano-services/deployments/blockfrost-backend-ingress.nix]: resolved, made it -blockfrost

**Review by gytis-ivaskevicius — APPROVED** [review]

**Pull request closed without merging**